src/compatibility.ts

```diff
@@ -74,3 +75,21 @@ export const requiredHubFlagWithDeprecations = requiredHubFlag({
   aliases: ['targetdevhubusername'],
   deprecateAliases: true,
 });
+
+/**
+ * @deprecated
+ */
+export const arrayWithDeprecation = (options: Partial<Omit<OptionFlagProps, 'multiple' | 'parse'>>) =>
+  Flags.string({
+    // populate passed options
+    ...options,
+    // overlay those options we own
+    multiple: true,
+    parse: async (input: string) => {
+      const inputParts = input.split(',').map((i) => i.trim());
+      if (inputParts.length > 1) {
+        await Lifecycle.getInstance().emit('warning', messages.getMessage('warning.arrayInputFormat'));
```

Comment:
```suggestion
        await Lifecycle.getInstance().emitWarning(messages.getMessage('warning.arrayInputFormat'));
```

Reply:
it's the same, but just more specific.

https://github.com/forcedotcom/sfdx-core/blob/02e51c10d113f80385c82996595b4e77a9710d6b/src/lifecycleEvents.ts#L182